lucille-core/src/main/java/com/kmwllc/lucille/core/Document.java

```diff
@@ -669,6 +669,36 @@ default void setNestedJson(String[] pathSegments, JsonNode value) {
     setField(pathSegments[0], node);
   }
 
+  default void removeNestedJson(String path) {
+    String[] parts = path.split("\\.");
+
+    if (parts.length == 0) {
+      return;
+    }
+    if (parts.length == 1) {
+      removeField(parts[0]);
+      return;
+    }
+
+    JsonNode parent = getNestedJson(String.join(".",
```

Comment:
should use the version of getNestedJson that takes String[] so we can avoid having to do a String.join here